### ISSUE TITLE
some problem about Genome doublings

### ISSUE THREAD

**Issue by LongpanICR** [ShixiangWang/DoAbsolute#23]
Dear Shixiang,
    thank you very much for creating a useful wrapper to run Absolute. However, I found there may be a problem on colunm "genome doubling" in the final table file. For instance, all the results are 0 in TCGA-LIHC cohort, which menas that there is no genome doubling in this cohort. In fact, it's not ture. Also, from original ABSOLUTE result, we can see nearly 30% of tumor have whole genome doubling. I am not sure if there are some problems when you adapted ABSOLUTE.

best regards,
Long PAN

**Comment by ShixiangWang** [comment]
@LongpanUPC Thanks. I basically don't modify any code regarding the computation. Could you share the ABSOLUTE version and source you used?

Go further, if you have more time, could you help me check the difference between results from the ABSOLUTE you are using and the ABSOLUTE inside the this repository (https://github.com/ShixiangWang/DoAbsolute/blob/master/inst/extdata/ABSOLUTE_1.0.6.tar.gz) with same ABSOLUTE commands and parameters? This can clarify the position with problem.

**Comment by LongpanICR** [comment]
@ShixiangWang ,Hi Shixiang, Sorry for late reply. 
In fact, I just compared the result from TCGA pancan project (https://gdc.cancer.gov/about-data/publications/pancanatlas, ABSOLUTE purity/ploidy file - [TCGA_mastercalls.abs_tables_JSedit.fixed.txt](http://api.gdc.cancer.gov/data/4f277128-f793-4354-a13d-30cc7fe9f6b5)) with the results generated by Doabsolute (https://zenodo.org/record/5751839#.YvuACXbP2Uk). Also, I used Doabsolute to estimate the ploidy/purity on another liver cancer cohort. Again, all the results are 0 in Genome doubling column.
  Your advice is useful and I will test the results according to your advice.

thank you very much!

**Comment by ShixiangWang** [comment]
@LongpanUPC Thank you in advance :). Currently, I have no idea why there is so much difference.

**Comment by Yunuuuu** [comment]
This maybe due to the conflicts between ubuntu openblas or Intel-blas and  ABSOLUTE.

I have integrated ABSOLUTE for personal usage in https://github.com/Yunuuuu/biomisc/blob/main/R/run_absolute.R which is modified from `DoAbsolute`, If I used future to implement parallel process, the results will be strange which I have reported in future issue
https://github.com/HenrikBengtsson/future/issues/649. 

I finally decided to run in sequential normal process, as ABSOLUTE can utilize multiple cores though I cannot have much power to adjust to the core number. 
![image](https://user-images.githubusercontent.com/19575010/222711998-9b1a4659-f959-4cb1-b972-52da2fe4c3ed.png)
